widget/embedded/src/containers/QuoteInfo/QuoteInfo.tsx

```diff
@@ -45,6 +88,213 @@ export function QuoteInfo(props: PropTypes) {
+      const stepHasWarning =
+        warning?.type === QuoteWarningType.INSUFFICIENT_SLIPPAGE &&
+        warning.recommendedSlippages?.has(index);
+
+      if (stepHasError) {
+        stepState = 'error';
+      } else if (stepHasWarning) {
+        stepState = 'warning';
+      }
+
+      let alertTitle = stepHasError
+        ? i18n.t('Slippage Error')
+        : i18n.t('Slippage Warning');
+
+      if (error?.type === QuoteErrorType.BRIDGE_LIMIT) {
+        alertTitle = error?.recommendation;
+      }
+
+      return {
+        swapper: {
+          displayName: getSwapperDisplayName(swap.swapperId, swappers),
+          image: swap.swapperLogo,
+        },
+        from: {
+          token: { displayName: swap.from.symbol, image: swap.from.logo },
+          chain: {
+            displayName: getBlockchainShortNameFor(
+              swap.from.blockchain,
+              blockchains
+            ),
+            image: swap.from.blockchainLogo,
+          },
+          price: {
+            value:
+              index === 0
+                ? numberToString(
+                    inputAmount,
+                    TOKEN_AMOUNT_MIN_DECIMALS,
+                    TOKEN_AMOUNT_MAX_DECIMALS
+                  )
+                : numberToString(
+                    swap.fromAmount,
+                    TOKEN_AMOUNT_MIN_DECIMALS,
+                    TOKEN_AMOUNT_MAX_DECIMALS
+                  ),
+            usdValue: numberToString(
+              (swap.from.usdPrice ?? 0) * parseFloat(swap.fromAmount),
+              USD_VALUE_MIN_DECIMALS,
+              USD_VALUE_MAX_DECIMALS
+            ),
+            realValue: formatTooltipNumbers(
+              index === 0 ? inputAmount : swap.fromAmount
+            ),
+            realUsdValue: formatTooltipNumbers(
+              new BigNumber(swap.from.usdPrice ?? 0).multipliedBy(
+                swap.fromAmount
+              )
+            ),
+          },
+        },
+        to: {
+          token: { displayName: swap.to.symbol, image: swap.to.logo },
+          chain: {
+            displayName: getBlockchainShortNameFor(
+              swap.to.blockchain,
+              blockchains
+            ),
+            image: swap.to.blockchainLogo,
+          },
+          price: {
+            value: numberToString(
+              swap.toAmount,
+              TOKEN_AMOUNT_MIN_DECIMALS,
+              TOKEN_AMOUNT_MAX_DECIMALS
+            ),
+            usdValue: numberToString(
+              (swap.to.usdPrice ?? 0) * parseFloat(swap.toAmount),
+              USD_VALUE_MIN_DECIMALS,
+              USD_VALUE_MAX_DECIMALS
+            ),
+            realValue: formatTooltipNumbers(swap.toAmount),
+            realUsdValue: formatTooltipNumbers(
+              new BigNumber(swap.to.usdPrice ?? 0).multipliedBy(swap.toAmount)
+            ),
+          },
+        },
+        state: stepState,
+        alerts:
+          stepHasError || stepHasWarning ? (
+            <FooterStepAlarm>
```

Comment:
Can we have a helper file to handle these jsx there?
Does it need a key (because of the map)?

Reply:
We decided to have a separate task for refactoring Quote components,
I think this case can be handled there as well.